app/src/main/java/net/sourceforge/opencamera/ui/RecSyncView.java

```diff
@@ -61,87 +74,71 @@ public RecSyncView(Context context) {
             Log.d(TAG, "small_screen: " + small_screen);
         }
 
-        final Preview preview = main_activity.getPreview();
+        button_sync_settings = addButton(R.string.sync_settings_unlocked, view -> {
+            if (MyDebug.LOG) {
+                Log.d(TAG, "clicked to buttonSyncSettings");
+            }
 
-        // TODO if (support recSync)
+            main_activity.clickedSyncSettings();
+        });
+
+        button_sync_devices = addButton(R.string.align_phases, view -> {
+            if (MyDebug.LOG) {
+                Log.d(TAG, "clicked to buttonAlignPhases");
+            }
+
+            main_activity.clickedAlignPhases();
+        });
+    }
+
+    public int getTotalWidth() {
+        return (int) (total_width_dp * scale + 0.5f); // convert dps to pixels;
+    }
+
+    public void updateSyncSettingsButton() {
+        button_sync_settings.setText(
+                software_sync_controller.isSettingsBroadcasting() ? R.string.sync_settings_locked : R.string.sync_settings_unlocked
+        );
+    }
+
+    private CheckBox addCheckbox(MainActivity context, @StringRes int text, boolean is_checked, CompoundButton.OnCheckedChangeListener listener) {
+        CheckBox check_box = new CheckBox(context);
+        check_box.setText(getResources().getString(text));
+        check_box.setTextSize(TypedValue.COMPLEX_UNIT_DIP, standard_text_size_dip);
+        check_box.setTextColor(Color.WHITE);
 
-        // checkbox switch recSync
-        CheckBox checkBox = new CheckBox(main_activity);
-        checkBox.setText(getResources().getString(R.string.preference_switch_rec_sync));
-        checkBox.setTextSize(TypedValue.COMPLEX_UNIT_DIP, standard_text_size_dip);
-        checkBox.setTextColor(Color.WHITE);
         {
             // align the checkbox a bit better
             LayoutParams params = new LayoutParams(
                     LayoutParams.MATCH_PARENT,
                     LayoutParams.MATCH_PARENT
             );
             final int left_padding = (int) (10 * scale + 0.5f); // convert dps to pixels
```

Comment:
all such convertions have to be in utils

Reply:
What is you mean?

Reply:
create some static class like Utils and move such convertions to it as static functions

Reply:
if we create Utils it will have only a single function and it will be used in one place. Do we need this class?

Reply:
I think that such convertions will be used a lot and it is a good idea to have such util method. Also usage of such conventions isn't the main task of this view, so extracting will be good for single responsibility principle

Reply:
Also check for small screen wil be in this class, for example